src/OpenTelemetry.Instrumentation.AspNetCore/MeterProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="MeterProviderBuilderExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Instrumentation.AspNetCore;
+using OpenTelemetry.Metrics;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Extension methods to simplify registering of ASP.NET Core request instrumentation.
+    /// </summary>
+    public static class MeterProviderBuilderExtensions
+    {
+        /// <summary>
+        /// Enables the incoming requests automatic data collection for ASP.NET Core.
+        /// </summary>
+        /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
+        /// <param name="configureAspNetCoreInstrumentationOptions">ASP.NET Core Request configuration options.</param>
+        /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        public static MeterProviderBuilder AddAspNetCoreInstrumentation(
+            this MeterProviderBuilder builder,
+            Action<AspNetCoreInstrumentationOptions> configureAspNetCoreInstrumentationOptions = null)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            // TODO: Currently these options are unused for metrics.
+            // Do we need separate options for metrics, or will the same options used for traces make sense for metrics?
```

Comment:
I think having one options across traces/logs/metrics are more friendly for the user.

FYI - I think some of the exporters will move towards that direction (e.g. a single ConsoleExporter targeting logs/traces/metrics)
https://github.com/open-telemetry/opentelemetry-dotnet/blob/5c855523106c0dc1271abc6a78cf9aeba88a7446/src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs#L23

Reply:
What are some of the possible things which can be configured in metricOptions? Do we need to allow user to specify labels? Or we always get all the labels as per semantic convention, and users can configure metric Views to drop any labels they dont care.

Reply:
There's nothing I'm aware of currently in the spec regarding the ability to configure labels to include/exclude. The concept of metric views is still in a proposed state https://github.com/open-telemetry/oteps/pull/89.

For the purpose of this work, it may just make sense to remove the ability to specify options for now. That is, the [`AspNetCoreInstrumentationOptions`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/874f9d78633fd74ae29a77c739ba69743f17802a/src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentationOptions.cs) currently has options that are not really yet applicable in the context of metrics. I think the one thing that makes sense in the `Filter` delegate to suppress capturing metrics based on certain criteria.